lib/launcher.js

```diff
@@ -47,13 +47,15 @@ class Launcher {
             return reporter
         }
 
-        for (let reporterType of config.reporter) {
-            try {
-                let Reporter = require(`wdio-${reporterType}-reporter`)
-                reporter.add(new Reporter(reporter, config))
-            } catch (e) {
-                throw new Error(`reporter "wdio-${reporterType}-reporter" is not installed. Error: ${e.stack}`)
+        for (let Reporter of config.reporter) {
+            if (typeof Reporter === 'string') {
+                try {
+                    Reporter = require(`wdio-${Reporter}-reporter`)
+                } catch (e) {
+                    throw new Error(`reporter "wdio-${Reporter}-reporter" is not installed. Error: ${e.stack}`)
+                }
             }
+            reporter.add(new Reporter(reporter, config))
```

Comment:
I don't think this is the right way we should implement this. What if `Reporter` has a spelling issue and is still a string? How about we do an `else if` and check if typeof `reporterType` is function?